### ISSUE TITLE
Missing docs for save points

### ISSUE THREAD

**Issue by jsumners** [knex/knex#5876]
https://github.com/tgriesser/knex/issues/2213#issuecomment-327595596

It seems the documentation around how passing a `transaction` instance around will work with regard to save points.

**Comment by mercmobily** [comment]
You're correct that the documentation around savepoints and nested transactions needs improvement. Looking at the referenced issue #2213, there's clearly confusion about how passing transaction instances around works with regard to savepoints.

**Current documentation:**

The current documentation does mention savepoints in a few places:

- The main index page mentions "transaction support (with savepoints)"
- There's a note that "Amazon Redshift does not support savepoints in transactions"
- The changelog mentions "Nested transactions automatically become savepoints, with `commit` & `rollback` releasing or rolling back the current savepoint"

However, as you and elhigu noted, there's no dedicated section explaining how savepoints work or how to use them.

**What the documentation should cover:**

As elhigu mentioned in the comments, the documentation should:

1. Explain that nested transactions are supported
2. Note that some database dialects implement nested transactions using savepoints (but not force all dialects to expose savepoints directly)
3. Show examples of how nested transactions work in practice
4. Explain what happens when you commit or rollback a nested transaction
5. Clarify which databases support this feature and which don't

The focus should be on the nested transaction feature rather than the implementation detail of savepoints, as elhigu suggested. This is a higher-level abstraction that's more useful for developers.

Example that should be in the docs:
```js
await knex.transaction(async (trx) => {
  await trx('accounts').insert({name: 'Account 1'});

  // This creates a savepoint in databases that support it
  await trx.transaction(async (nestedTrx) => {
    await nestedTrx('logs').insert({action: 'nested operation'});
    // This only rolls back the nested transaction/savepoint
    throw new Error('Rollback nested');
  }).catch(() => {});

  // This still succeeds
  await trx('accounts').insert({name: 'Account 2'});
});
```

**Related Issues:**
- #2213 - Discussion about how transaction instances work with savepoints
